app/test/privacy_policy/table_of_contents/toc_currently_reading_test.dart

```diff
@@ -0,0 +1,833 @@
+  return DocumentSection(
+      DocumentSectionId(id), id, subsections.toIList() ?? const IListConst([]));
+}
+
+DocumentSectionHeadingPosition _headingPosition(
+  String sectionId, {
+  @required double itemLeadingEdge,
+  @required double itemTrailingEdge,
+}) {
+  return DocumentSectionHeadingPosition(
+    DocumentSectionId(sectionId),
+    itemLeadingEdge: itemLeadingEdge,
+    itemTrailingEdge: itemTrailingEdge,
+  );
+}
+
+void main() {
+  group('the table of contents', () {
+    ValueNotifier<List<DocumentSectionHeadingPosition>> visibleSections;
+
+    setUp(() {
+      visibleSections = ValueNotifier<List<DocumentSectionHeadingPosition>>([]);
+      FlutterError.onError = (e) => fail(e.toString());
+    });
+
+    TestCurrentlyReadingController _createController(
+      List<DocumentSection> sections, {
+      double threshold = 0.1,
+      String endSection,
+    }) {
+      return TestCurrentlyReadingController(
+        sections,
+        visibleSections,
+        threshold: CurrentlyReadThreshold(threshold),
+        endSection: endSection != null ? DocumentSectionId(endSection) : null,
+      );
+    }
+
+    test(
+        'doesnt mark any section as currently read if none are or have been visible on the page',
+        () {
+      final sections = [
+        _section('foo'),
+      ];
+
+      final controller = _createController(sections);
+
+      expect(controller.currentlyReadSection, null);
+    });
+
+    test(
+        'Doesnt mark a section as currently read when the top of the section is located before the threshold',
+        () {
+      final sections = [
+        _section('foo'),
+      ];
+
+      final controller = _createController(sections, threshold: 0.1);
+
+      visibleSections.value = [
+        _headingPosition(
+          'foo',
+          itemLeadingEdge: 0.11,
+          itemTrailingEdge: 0.2,
+        ),
+      ];
+
+      expect(controller.currentlyReadSection, null);
+    });
+
+    test(
+        'Marks a section as currently read when the top of the section touches the threshold',
+        () {
+      final sections = [
+        _section('foo'),
+      ];
+
+      final controller = _createController(sections, threshold: 0.1);
+
+      visibleSections.value = [
+        _headingPosition(
+          'foo',
+          itemLeadingEdge: 0.1,
+          itemTrailingEdge: 0.15,
+        ),
+      ];
+
+      expect(controller.currentlyReadSection, 'foo');
+    });
+
+    test(
+        'Marks a section as currently read when the the section intersects the threshold',
+        () {
+      final sections = [
+        _section('foo'),
+      ];
+
+      final controller = _createController(sections, threshold: 0.1);
+
+      visibleSections.value = [
+        _headingPosition(
+          'foo',
+          itemLeadingEdge: 0.05,
+          itemTrailingEdge: 0.15,
+        ),
+      ];
+
+      expect(controller.currentlyReadSection, 'foo');
+    });
+    test(
```

Comment:
```suggestion

    test(
```

Add new line after test